src/lang-promql/parser/matcher.test.ts

```diff
@@ -0,0 +1,112 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2020 The Prometheus Authors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import { EqlRegex, EqlSingle, Neq, NeqRegex } from 'lezer-promql';
+import { labelMatchersToString } from './matcher';
+import { Matcher } from '../types/matcher';
+import chai from 'chai';
+
+describe('labelMatchersToString test', () => {
+  const testCases = [
+    {
+      title: 'metric_name',
+      metricName: 'metric_name',
+      matchers: [] as Matcher[],
+      result: 'metric_name',
+    },
+    {
+      title: 'metric_name 2',
+      metricName: 'metric_name',
+      matchers: undefined,
+      result: 'metric_name',
+    },
+    {
+      title: 'metric_name{}',
+      metricName: 'metric_name',
+      matchers: [
+        {
+          type: EqlSingle,
+          name: 'LabelName',
+          value: '',
+        },
+      ] as Matcher[],
+      result: 'metric_name{}',
+    },
+    {
+      title: 'sum{LabelName!="LabelValue"}',
+      metricName: 'sum',
+      matchers: [
+        {
+          type: Neq,
+          name: 'LabelName',
+          value: 'LabelValue',
+        },
+      ] as Matcher[],
+      result: 'sum{LabelName!="LabelValue"}',
+    },
+    {
+      title: 'rate{labelName=~"label.+"}',
+      metricName: 'rate',
+      matchers: [
+        {
+          type: EqlSingle,
+          name: 'LabelName',
+          value: '',
+        },
+        {
+          type: EqlRegex,
+          name: 'LabelName',
+          value: 'label.+',
+        },
+      ] as Matcher[],
+      result: 'rate{LabelName=~"label.+"}',
```

Comment:
Also, the result here should include both of the `labelName` matchers right? It's valid to do something like `rate{LabelName="label1", labelName="label1"}`

Reply:
wait sorry I realized it's being removed because it's an empty string :)